mgr/dashboard: Nvme mTLS support and  service name changes

afreen23 opened the pull request:
[mgr/dashboard: Adding group and pool name to service name](https://github.com/ceph/ceph/commit/551b6b2924933f5a43acc1aea98978e41193cfab) 

- Pre-populating the service name field with the format `nvmeof.<pool_name>.<group_name>`.
- This can be changed by user but by default this value will be there.
- This will help user to quickly fill form and proceed hence improving usability.
- cephadm also uses this format as of now this convention so it will make UI aligned with CLI experience

- updates unit tests to improve coverage

- hides`count` values as that is not needed for 'nvmeof' only hosts and labels required.

Fixes https://tracker.ceph.com/issues/68065

[mgr/dashboard: Add mTLS support](https://github.com/ceph/ceph/commit/4db35e8a4f8884d93874cd06da826ba0746974dd) 

- enables mTLS support from dashboard
- adds unit tests related to mTLS support
- can enable mTLS
- can disable mTLS
- inlcuded refactoring from prev commit

## Nvmeof service without mTLS
[nvme-without-mtls.webm](https://github.com/user-attachments/assets/f30ea81d-b5eb-46e2-8577-c878ce11721c)


## Nvmeof service with mTLS
[nvme-with-mtls.webm](https://github.com/user-attachments/assets/64e96ec8-a57a-4e0e-b900-63abc3fe233a)


## For testing this PR:
* `ceph osd pool create nvmeof_pool01`
* `rbd pool init nvmeof_pool01`

* From dashboard create a nvmeof service via service form
* service ID /group/pool values prepoluated
* enable encryption 
* fill in dummy cert values
* select hosts
* `count` should not be visible
* create service

* edit service created above
* uncheck encryption and proceed.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

nizamial09 commented on src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.html:
```diff
@@ -292,7 +292,7 @@
 
         <!-- Count -->
-        <div *ngIf="!serviceForm.controls.unmanaged.value"
+        <div *ngIf="!serviceForm.controls.unmanaged.value && serviceForm.controls.service_type.value !== 'nvmeof'"
```
why are we hiding the count in the case of nvmeof?

afreen23 replied:
It was suggested in discussions with nvme team and Paul.
This is of no use actually. Quoting what @pcuzner mentioned earlier on mockups:
>  count makes no sense here - at least for 8.0. one host will only support one gateway daemon. count is used to scale up stateless daemons like rgw

I missed to remove this in previous implementation

afreen23 commented:
Pool tests failure: due to [this](https://github.com/ceph/ceph/pull/59805/files#diff-ee5df2c612afdd2344bd553a9451a04f53ecfa88145cb4e82e5e407d0068c5cd)

```
 ● PoolDetailsComponent › Pool details tabset › Updates of shown data › should update shown data on change

    expect(received).toEqual(expected) // deep equality

    - Expected  - 3
    + Received  + 0

    @@ -1,10 +1,7 @@
      Object {
        "poolDetails": Object {
    -     "application_metadata": Array [
    -       "rbd",
    -     ],
          "pg_num": 256,
          "pg_num_target": 256,
          "pg_placement_num": 256,
          "pg_placement_num_target": 256,
          "pool": 2,

      117 |           expect(expected).toEqual(oldData);
      118 |         }
    > 119 |         expect(expected).toEqual(newData);
          |                          ^
      120 |       };
      121 |
      122 |       it('should update shown data on change', () => {

      at expectedChange (src/app/ceph/pool/pool-details/pool-details.component.spec.ts:119:26)
   ```

afreen23 commented:
jenkins test api

afreen23 commented:
jenkins test make check

afreen23 commented:
> Pool tests failure: due to [this](https://github.com/ceph/ceph/pull/59805/files#diff-ee5df2c612afdd2344bd553a9451a04f53ecfa88145cb4e82e5e407d0068c5cd)
> 
> ```
>  ● PoolDetailsComponent › Pool details tabset › Updates of shown data › should update shown data on change
> 
>     expect(received).toEqual(expected) // deep equality
> 
>     - Expected  - 3
>     + Received  + 0
> 
>     @@ -1,10 +1,7 @@
>       Object {
>         "poolDetails": Object {
>     -     "application_metadata": Array [
>     -       "rbd",
>     -     ],
>           "pg_num": 256,
>           "pg_num_target": 256,
>           "pg_placement_num": 256,
>           "pg_placement_num_target": 256,
>           "pool": 2,
> 
>       117 |           expect(expected).toEqual(oldData);
>       118 |         }
>     > 119 |         expect(expected).toEqual(newData);
>           |                          ^
>       120 |       };
>       121 |
>       122 |       it('should update shown data on change', () => {
> 
>       at expectedChange (src/app/ceph/pool/pool-details/pool-details.component.spec.ts:119:26)
> ```

Fixed and rebased after SSO PR
